ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/TermCompiler.java

```diff
@@ -0,0 +1,559 @@
+    return visitIllegalState(node);
+  }
+
+  @Override
+  public Term visitWhenClause(final WhenClause whenClause, final Void context) {
+    return visitIllegalState(whenClause);
+  }
+
+  @Override
+  public Term visitInPredicate(
+      final InPredicate inPredicate,
+      final Void context
+  ) {
+    return visitUnsupported(inPredicate);
+  }
+
+  @Override
+  public Term visitInListExpression(
+      final InListExpression inListExpression, final Void context
+  ) {
+    return visitUnsupported(inListExpression);
+  }
+
+  @Override
+  public Term visitTimestampLiteral(
+      final TimestampLiteral node, final Void context
+  ) {
+    return BasicTerms.of(node.getValue());
+  }
+
+  @Override
+  public Term visitTimeLiteral(
+      final TimeLiteral timeLiteral,
+      final Void context
+  ) {
+    return visitUnsupported(timeLiteral);
+  }
+
+  @Override
+  public Term visitSimpleCaseExpression(
+      final SimpleCaseExpression simpleCaseExpression, final Void context
+  ) {
+    return visitUnsupported(simpleCaseExpression);
+  }
+
+  @Override
+  public Term visitBooleanLiteral(
+      final BooleanLiteral node,
+      final Void context
+  ) {
+    return BasicTerms.of(node.getValue());
+  }
+
+  @Override
+  public Term visitStringLiteral(final StringLiteral node, final Void context) {
+    return BasicTerms.of(node.getValue());
+  }
+
+  @Override
+  public Term visitDoubleLiteral(final DoubleLiteral node, final Void context) {
+    return BasicTerms.of(node.getValue());
+  }
+
+  @Override
+  public Term visitDecimalLiteral(
+      final DecimalLiteral decimalLiteral,
+      final Void context
+  ) {
+    final SqlType sqlType = DecimalUtil.fromValue(decimalLiteral.getValue());
+    return BasicTerms.of(decimalLiteral.getValue(), sqlType);
+  }
+
+  @Override
+  public Term visitNullLiteral(final NullLiteral node, final Void context) {
+    return BasicTerms.ofNull();
+  }
+
+  @Override
+  public Term visitLambdaExpression(
+      final LambdaFunctionCall lambdaFunctionCall, final Void context) {
+    return visitUnsupported(lambdaFunctionCall);
+  }
+
+  @Override
+  public Term visitLambdaVariable(
+      final LambdaVariable lambdaLiteral, final Void context) {
+    return visitUnsupported(lambdaLiteral);
+  }
+
+  @Override
+  public Term visitUnqualifiedColumnReference(
+      final UnqualifiedColumnReferenceExp node,
+      final Void context
+  ) {
+    final Column schemaColumn = schema.findValueColumn(node.getColumnName())
+        .orElseThrow(() ->
+            new KsqlException("Field not found: " + node.getColumnName()));
+
+    return ColumnReferenceTerm.create(schemaColumn.index(), schemaColumn.type());
+  }
+
+  @Override
+  public Term visitQualifiedColumnReference(
+      final QualifiedColumnReferenceExp node,
+      final Void context
+  ) {
+    throw new UnsupportedOperationException(
+        "Qualified column reference must be resolved to unqualified reference before codegen"
```

Comment:
nit: this is no longer codegen :) should this be an illegal state exception?

Reply:
Updated!